test/liblangutil/CharStream.cpp

```diff
@@ -48,6 +48,62 @@ BOOST_AUTO_TEST_CASE(test_fail)
 	);
 }
 
+#define testLineColumnToPositionTranslationFail(_line, _column, _text) \
+	do                                                                       \
+	{                                                                        \
+		auto const source = std::make_shared<CharStream>((_text), "source"); \
+		auto const actualPosition =                                          \
+			source->translateLineColumnToPosition((_line), (_column));       \
+		BOOST_CHECK_EQUAL(actualPosition.value_or(-1), -1);                  \
+	}                                                                        \
+	while (0)
+
+#define testLineColumnToPositionTranslation(_line, _column, _expectedPosition, _text) \
+	do                                                                       \
+	{                                                                        \
+		auto const source = std::make_shared<CharStream>((_text), "source"); \
+		auto const actualPosition =                                          \
+			source->translateLineColumnToPosition((_line), (_column));       \
+		BOOST_CHECK(actualPosition.has_value());                             \
+		BOOST_CHECK_EQUAL(*actualPosition, (_expectedPosition));             \
+	}                                                                        \
+	while (0)
+
+BOOST_AUTO_TEST_CASE(translateLineColumnToPosition)
+{
+	testLineColumnToPositionTranslationFail(0, 0, "");
```

Comment:
Is this what we want to have?

Reply:
I think you're right. we should not want that.

Reply:
I disagree. If you run `contentChanges` on an empty file, there should be a way to add text to it (apart from changing the whole file). Can you please check if the "line/colmn" thing is defined in the lsp specification somewhere?

Reply:
ok. I understand. I agree. :)

Reply:
quote from the spec:
> A position inside a document (see Position definition below) is expressed as a zero-based line and character offset. The offsets are based on a UTF-16 string representation.

with that example:
> So a string of the form a𐐀b the character offset of the character a is 0, the character offset of 𐐀 is 1 and the character offset of b is 3 since 𐐀 is represented using two code units in UTF-16.

I am not sure how urgent multi-codepoint graphemes are with respect to cursor positioning or if that can wait until after initial release. We would need a UTF-8 <-> UTF-16 conversion helper.

Reply:
So it has nothing about how newlines are interpreted and which columns are valid?